draft-ietf-tls-esni.md

```diff
@@ -1025,9 +999,32 @@ The backend server MUST NOT perform this operation if it negotiated TLS 1.2 or
 below. Note that doing so would overwrite the downgrade signal for TLS 1.3 (see
 {{RFC8446}}, Section 4.1.3).
 
-The "ech_is_inner" is expected to have an empty payload. If the payload is
-non-empty (i.e., the length of the "extension_data" field is non-zero) then the
-backend server MUST abort the handshake with an "illegal_parameter" alert.
+The payload of "encrypted_client_hello" is expected to be a `ClientECH` with
+`ClientECH.type` is `inner`. If this is not the case, the backend server MUST
+abort the handshake with an "illegal_parameter" alert.
+
+### Sending HelloRetryRequest {#backend-server-hrr}
+
+If the backend server sends HelloRetryRequest in response to the ClientHello, it
+confirms ECH acceptance as follows. It generates a message
+HelloRetryRequestECHConf equal to HelloRetryRequest, but with an
+"encrypted_client_hello" extension with as payload of 8 zero bytes. It then
+computes an 8-byte string
+
+~~~
+   hrr_accept_confirmation = HKDF-Expand-Label(0,
+      "hrr ech accept confirmation",
+      Transcript-Hash(
+          ClientHelloInner...HelloRetryRequestECHConf),
+      8)
+~~~
+
+where HKDF-Expand-Label, Transcript.Hash, and "0" are as defined above and
+ClientHelloInner...HelloRetryRequestECHConf denotes the sequence of handshake
```

Comment:
Please, work on improving the definition of `Transcript-Hash()`.  As I commented in the other PR, this definition is too loose.

Reply:
I'm sorry, Martin, I'm afraid I've lost the context here. Can you point me to your original comment, or recall the change that's needed?

Reply:
What did you think it meant? The syntax seems pretty clear (to me) -- it's a sequence of messages. Is the problem that the sequence of messages is not clear? Should we just be explicit and list all messages, rather than just assume the reader can figure it out? That is, instead of:

`ClientHelloInner...ServerHelloECHConf`

say

`ClientHelloInner || ServerHelloECHConf`

in the happy path, and, e.g., 

`ClientHelloInner1 || HelloRetryRequest || ClientHelloInner2 || ServerHelloECHConf`

in the HRR+ECH-accept path?